runtime/include/error.h

```diff
@@ -25,21 +25,26 @@
 
 extern int verbosity;
 
+// These functions eventually call out to chpl_exit_any, or try to perform a
+// filename index -> filename lookup, neither of which can be performed while
+// the runtime is being unit tested
 #ifndef CHPL_RT_UNIT_TEST
-void chpl_warning(const char* message, int32_t lineno, c_string filename);
-void chpl_error(const char* message, int32_t lineno, c_string filename);
+void chpl_warning(const char* message, int32_t lineno, int32_t filenameIdx);
+void chpl_error(const char* message, int32_t lineno, int32_t filenameIdx);
 void chpl_internal_error(const char* message);
 #else
+// Filename is now an int32_t index into a table that we are not going to have
+// while the runtime is in unit test mode, just print out the message instead
 #define chpl_warning(message, lineno, filename) \
     do {                                        \
       fflush(stdout);                           \
-      fprintf(stderr, "%s:%" PRId32 ": warning: %s\n", filename, lineno, message); \
+      fprintf(stderr, "warning: %s\n", message); \
```

Comment:
What will get passed here, any idea?  Could the line number still be useful, or are (lineno, filename) always (0, 0)?  I confess I don't know -- Michael added this unit test mode quite a while ago so he could test runtime changes without writing Chapel programs and I've never taken a hard look at how it really works.